spec/classes/puppet_agent_spec.rb

```diff
@@ -52,7 +52,7 @@
         end
 
         it do
-          should contain_puppet__config__agent('server').with_value('foo.example.com')
+          should contain_puppet__config__main('server').with_value('foo.example.com')
```

Comment:
Alright. I just moved them to puppet_config_spec.rb. I also did a little bit of restructuring, hope thats ok.